### DISCUSSION TITLE
Can we have a call to talk about filtering and Aerospike Expressions

### DISCUSSION THREAD

**Discussion by rbotzer** [aerospike/spring-data-aerospike#535] [Team Posts]
Hi. In [Aerospike version 5.2](https://www.aerospike.com/blog/aerospike5-2/) we released [Aerospike Expressions](https://www.aerospike.com/docs/guide/expressions/). Among other things Expressions are used to replace Predicate Expressions (PredExp) for filtering. I don't believe Spring Data Aerospike currently uses PredExp, though.

Filters can be used as a WHERE clause to determine whether to return rows from a scan or query, and also as a conditional for all key-value operations (put, get, delete, operate), batch-reads and record UDFs. 

I wanted to know if we could schedule a call to talk about Aerospike Expressions and the direction we're taking them. They're powerful and will get even more expressive in the next releases.

Another question I have is whether Spring Data for Aerospike supports multi-operation transactions on a single record. This is what the `operate` command does in Aerospike.

Please email me at ronen@aerospike.com if you can spare some time for a call.

Thanks,
Ronen

**Comment by Aloren** [comment]
Hi Ronen.

> I don't believe Spring Data Aerospike currently uses PredExp, though.

spring-data uses predexps for querying Aerospike. We can add support for the Aerospike expressions, I suppose.

> Another question I have is whether Spring Data for Aerospike supports multi-operation transactions on a single record. This is what the operate command does in Aerospike.

No, this is not supported. 

>Please email me at ronen@aerospike.com if you can spare some time for a call.

Ok.
